src/main/java/frc/robot/commands/ChargeCommandPigeon.java

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.trajectory.TrapezoidProfile.Constraints;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.SwerveSubsystem;
+
+public class ChargeCommandPigeon extends CommandBase {
+  Timer m_timer = new Timer();
+
+  ProfiledPIDController pidControlTheta;
+  double currentTheta;
+  double desiredTheta;
+
+  double pigeonValue;
+  double initPigeonValue;
+  boolean bWideSide;
+  double flipXDirection;
+
+  double PITCH_TOLERANCE = 10;
+
+  //@todo: tune these values
+  double TIME_FOR_REVERSING = 0.10;
+  double X_SPEED_STATE0 = 1.5;
+  double X_SPEED_STATE1 = 1.0;
+  double X_SPEED_STATE2 = 0.5;
```

Comment:
X_SPEED_STATE2 should be negative. It's supposed to be a quick backwards voltage to slow down the chassis.

Reply:
I didn't realize this was handled elsewhere.